cpu/stm32/include/periph/cpu_sdmmc.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2023 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup         cpu_stm32
+ * @{
+ *
+ * @file
+ * @brief           CPU specific definitions for SDIO/SDMMC for the STM32 family
+ *
+ * @author          Gunar Schorcht <gunar@schorcht.net>
+ */
+
+#ifndef PERIPH_CPU_SDMMC_H
+#define PERIPH_CPU_SDMMC_H
+
+#include <stdint.h>
+
+#include "periph/cpu_dma.h"
+#include "periph/cpu_gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* For F1, F2, F4 and L1 the SDMMC interface is called SDIO, define used
+ * symbols for source code compatibility */
+#if !defined(SDMMC1) && !DOXYGEN
+#define SDMMC_TypeDef   SDIO_TypeDef
+#endif
+
+/**
+ * @brief SDIO/SDMMC buffers alignment because of STM32 DMA/FIFO restrictions
+ */
+#define SDMMC_CPU_DMA_ALIGNMENT     4
+
+/**
+ * @brief SDIO/SDMMC buffer instantiation requirement for STM32
+ */
+#define SDMMC_CPU_DMA_REQUIREMENTS  __attribute__((aligned(SDMMC_CPU_DMA_ALIGNMENT)))
+
+/**
+ * @brief SDIO/SDMMC pin structure for STM32
+ */
+typedef struct {
+    gpio_t pin;             /**< GPIO pin */
+#ifndef CPU_FAM_STM32F1
+    gpio_af_t af;           /**< GPIO alternate function */
+#endif
+} sdmmc_pin_t;
+
+/**
+ * @brief SDIO/SDMCC peripheral configuration for STM32
+ *
+ * To use 1-bit bus width, define `dat1` to `dat3` as `GPIO_UNDEF`. 8-bit bus
+ * width with `dat4` to `dat7` is only available if the board provides
+ * feature `periph_sdmmc_8bit`.
+ */
+typedef struct {
+    SDMMC_TypeDef *dev;         /**< SDIO/SDMMC device */
+    uint8_t bus;                /**< APB/AHB bus used for SDIO/SDMMC peripheral */
```

Comment:
If you move `bus` to the end  of the struct you'll save 4 bytes of alignment gap

Reply:
But only 4 bytes of flash because of static const data.